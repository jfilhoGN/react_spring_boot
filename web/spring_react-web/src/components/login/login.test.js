import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import Login from './login';

describe('<Login />', () => {
        
    test('render component', () => {
        const wrapper = shallow(<Login/>)
        expect(wrapper).toMatchSnapshot();
    });

    test('values of state', () => {
        const wrapper = shallow(<Login/>);
        //expect(wrapper.state).toMatchSnapshot();
        //arrumar
    });

    test('input text email', () => {
        const wrapper = shallow(<Login />);
        wrapper.find('#email').simulate('change',{ target: { value: '12345'} });
        expect(wrapper.state().email).toEqual('12345');
    });
       
    test('input text senha', () => {
        const wrapper = shallow(<Login />);
        wrapper.find('#senha').simulate('change',{ target: { value: '12345'} });
        expect(wrapper.state().senha).toEqual('12345');
    });

    test('button login', () => {
        //arrumar    
    });

    test('button addFuncionario', () => {
        //arrumar
    });

    test('acceptFuncionario()', () => {
        //arrumar
    });

    // Ver as funções handleChangeEmail e
    // handleChangeSenha 
    // addClick precisam fazer teste

    test('handleSubmit()', () => {
        //arrumar    
    });


});