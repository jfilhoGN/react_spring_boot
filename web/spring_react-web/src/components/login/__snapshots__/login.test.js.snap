// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> render component 1`] = `
<div
  className="login-page"
>
  <form
    className="formLogin"
    onSubmit={[Function]}
  >
    <FormGroup
      bsClass="form-group"
      bsSize="large"
      controlId="email"
    >
      <ControlLabel
        bsClass="control-label"
        srOnly={false}
      >
        Email
      </ControlLabel>
      <FormControl
        autoFocus={true}
        bsClass="form-control"
        componentClass="input"
        id="email"
        onChange={[Function]}
        type="email"
      />
    </FormGroup>
    <FormGroup
      bsClass="form-group"
      bsSize="large"
      controlId="password"
    >
      <ControlLabel
        bsClass="control-label"
        srOnly={false}
      >
        Senha
      </ControlLabel>
      <FormControl
        bsClass="form-control"
        componentClass="input"
        id="senha"
        onChange={[Function]}
        type="password"
      />
    </FormGroup>
    <Button
      active={false}
      block={true}
      bsClass="btn"
      bsSize="large"
      bsStyle="default"
      disabled={false}
      onClick={[Function]}
      type="submit"
    >
      Login
    </Button>
    <br />
    <Button
      active={false}
      block={false}
      bsClass="btn"
      bsSize="large"
      bsStyle="default"
      disabled={false}
      onClick={[Function]}
      type="submit"
    >
      Criar Conta
    </Button>
  </form>
</div>
`;
